packages/menu/src/Menu/Menu.tsx

```diff
@@ -264,8 +266,8 @@ export const Menu = React.forwardRef<HTMLDivElement, MenuProps>(function Menu(
             {/* Need to stop propagation, otherwise Menu will closed automatically when clicked */}
             {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events*/}
             <ul
-              data-testid={lgIds.root}
-              {...rest}
+              data-testid={(rest as any)['data-testid'] || lgIds.root}
```

Comment:
We should extract `testIdProp` from `rest` in the component props

```suggestion
              data-testid={testIdProp || lgIds.root}
```

Reply:
fixed!